[
    {
        "content": "<p>After closely following the installation instructions in step 1, the installation fails in step 2 giving the error \"virtualenv: error: unrecognized arguments: --site-packages\". It appears like the virtual environment is unable to start (see screenshot). Running on a clean ubuntu 18.04, up to date, lxc instance. <a href=\"/user_uploads/2/57/ftinQQHAefuhWk4Cp9ceP4Cm/image.png\">image.png</a></p>\n<div class=\"message_inline_image\"><a href=\"/user_uploads/2/57/ftinQQHAefuhWk4Cp9ceP4Cm/image.png\" title=\"image.png\"><img src=\"/user_uploads/2/57/ftinQQHAefuhWk4Cp9ceP4Cm/image.png\"></a></div>",
        "id": 902231,
        "sender_full_name": "Pita Bread",
        "timestamp": 1591932062
    },
    {
        "content": "<p>Note that this is at the very bottom of the install.log</p>",
        "id": 902234,
        "sender_full_name": "Pita Bread",
        "timestamp": 1591932151
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"16236\">@Pita Bread</span>  Can you tell which version of virtualenv you are using?</p>\n<div class=\"codehilite\"><pre><span></span><code>virtualenv --version\n</code></pre></div>",
        "id": 902247,
        "sender_full_name": "Arpit Sharma",
        "timestamp": 1591935169
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"7\">@Tim Abbott</span>  Still  I think we can just remove <code>--site-packages</code> since it is <a href=\"https://github.com/pypa/virtualenv/blame/legacy/docs/reference.rst#L100\">depreciated </a></p>\n<blockquote>\n<p>--no-site-packages    DEPRECATED. Retained only for backward compatibility.<br>\n                        Not having access to global site-packages is now the<br>\n                        default behavior.</p>\n</blockquote>",
        "id": 902251,
        "sender_full_name": "Arpit Sharma",
        "timestamp": 1591935884
    },
    {
        "content": "<p>I'm not entirely sure what code is running here.</p>",
        "id": 902255,
        "sender_full_name": "Tim Abbott",
        "timestamp": 1591936278
    },
    {
        "content": "<p>Grepping our codebase, I don't think we use that option anywhere I can easily see.</p>",
        "id": 902256,
        "sender_full_name": "Tim Abbott",
        "timestamp": 1591936306
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"16236\">@Pita Bread</span> make sure you are using latest version of vagrant and virtualbox and are on zulip/zulip master.</p>",
        "id": 902286,
        "sender_full_name": "Aman (amanagr)",
        "timestamp": 1591939843
    },
    {
        "content": "<p><a href=\"/user_uploads/2/18/aUVgjq12sdN0VQHaFOoFZ72d/image.png\">image.png</a> Version 20.0.21</p>\n<div class=\"message_inline_image\"><a href=\"/user_uploads/2/18/aUVgjq12sdN0VQHaFOoFZ72d/image.png\" title=\"image.png\"><img src=\"/user_uploads/2/18/aUVgjq12sdN0VQHaFOoFZ72d/image.png\"></a></div>",
        "id": 902680,
        "sender_full_name": "Pita Bread",
        "timestamp": 1591970728
    },
    {
        "content": "<p>Should be on the master branch, it does not appear like vagrant was installed. Going to install it and update virtual env and try again.</p>",
        "id": 902683,
        "sender_full_name": "Pita Bread",
        "timestamp": 1591971170
    },
    {
        "content": "<p>Still getting the same error <a href=\"/user_uploads/2/35/885yHdl5vxyA54IVZodu5p3S/image.png\">image.png</a></p>\n<div class=\"message_inline_image\"><a href=\"/user_uploads/2/35/885yHdl5vxyA54IVZodu5p3S/image.png\" title=\"image.png\"><img src=\"/user_uploads/2/35/885yHdl5vxyA54IVZodu5p3S/image.png\"></a></div>",
        "id": 902721,
        "sender_full_name": "Pita Bread",
        "timestamp": 1591972924
    },
    {
        "content": "<p>Running the command without <code>--no-site-packages</code> appears to work <a href=\"/user_uploads/2/1a/eHaJt_EbfF-jhygb497ljiJz/image.png\">image.png</a></p>\n<div class=\"message_inline_image\"><a href=\"/user_uploads/2/1a/eHaJt_EbfF-jhygb497ljiJz/image.png\" title=\"image.png\"><img src=\"/user_uploads/2/1a/eHaJt_EbfF-jhygb497ljiJz/image.png\"></a></div>",
        "id": 902744,
        "sender_full_name": "Pita Bread",
        "timestamp": 1591973811
    },
    {
        "content": "<p>Interesting that it is using python 2.7 <a href=\"/user_uploads/2/69/ST9oHAHe7zS7ooHxhyw-w5YJ/image.png\">image.png</a></p>\n<div class=\"message_inline_image\"><a href=\"/user_uploads/2/69/ST9oHAHe7zS7ooHxhyw-w5YJ/image.png\" title=\"image.png\"><img src=\"/user_uploads/2/69/ST9oHAHe7zS7ooHxhyw-w5YJ/image.png\"></a></div>",
        "id": 902756,
        "sender_full_name": "Pita Bread",
        "timestamp": 1591974907
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"16236\">@Pita Bread</span> can you explain precisely what you're trying to do to install Zulip?  I.e. what top command are you running?</p>",
        "id": 903120,
        "sender_full_name": "Tim Abbott",
        "timestamp": 1591997274
    },
    {
        "content": "<p>I am trying to install zulip for the first time using this command <a href=\"/user_uploads/2/9a/PG-lByAreS47B_J0I7CLcvpp/image.png\">image.png</a></p>\n<div class=\"message_inline_image\"><a href=\"/user_uploads/2/9a/PG-lByAreS47B_J0I7CLcvpp/image.png\" title=\"image.png\"><img src=\"/user_uploads/2/9a/PG-lByAreS47B_J0I7CLcvpp/image.png\"></a></div>",
        "id": 903164,
        "sender_full_name": "Pita Bread",
        "timestamp": 1591999649
    },
    {
        "content": "<p>Step 1 of the guide went without any issues <a href=\"https://zulip.readthedocs.io/en/stable/production/install.html#installer-details\">https://zulip.readthedocs.io/en/stable/production/install.html#installer-details</a></p>",
        "id": 903166,
        "sender_full_name": "Pita Bread",
        "timestamp": 1591999677
    },
    {
        "content": "<p>More specifically it is the second part <code>./zulip-server-*/scripts/setup/install --certbot \\\n    --email=YOUR_EMAIL --hostname=YOUR_HOSTNAME</code> that fails. I did change the hostname and email address to the correct values.</p>",
        "id": 903176,
        "sender_full_name": "Pita Bread",
        "timestamp": 1591999825
    },
    {
        "content": "<p>Can you post the full output of that <code>install</code> script ( <code>/var/log/zulip/install.log</code> has it)?</p>",
        "id": 903190,
        "sender_full_name": "Tim Abbott",
        "timestamp": 1592000704
    },
    {
        "content": "<p>I'm guessing things go wrong well before the screenshots you're posting.</p>",
        "id": 903191,
        "sender_full_name": "Tim Abbott",
        "timestamp": 1592000713
    },
    {
        "content": "<p><a href=\"/user_uploads/2/cb/ko30V1BAKSGAAow_LmIGdAYR/install.log\">install.log</a></p>",
        "id": 903326,
        "sender_full_name": "Pita Bread",
        "timestamp": 1592009500
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"7\">@Tim Abbott</span> That should be the correct log file.</p>",
        "id": 903330,
        "sender_full_name": "Pita Bread",
        "timestamp": 1592009528
    },
    {
        "content": "<p>It looks like the rest of the install went smoothly</p>",
        "id": 903369,
        "sender_full_name": "Pita Bread",
        "timestamp": 1592010636
    },
    {
        "content": "<p>I got it to work! I created a VM (not an lxc) this time and make sure ALL of the packages were updated, and the installing script worked fine.</p>",
        "id": 903404,
        "sender_full_name": "Pita Bread",
        "timestamp": 1592014832
    },
    {
        "content": "<p>Aha, I understand the problem now <span class=\"user-mention\" data-user-id=\"16236\">@Pita Bread</span>!  It appears the error is coming from <code>certbot</code>; apparently their installer fails in LXC.</p>",
        "id": 903452,
        "sender_full_name": "Tim Abbott",
        "timestamp": 1592023038
    },
    {
        "content": "<p>I would open an issue to do something about that, but we're actually removing use of that certbot installer in favor of the apt package in our release coming out in 2 weeks (and doing so because certbot is no longer maintaining that install mechanism), so there's nothing to fix in master.</p>",
        "id": 903453,
        "sender_full_name": "Tim Abbott",
        "timestamp": 1592023090
    }
]